### PR TITLE
[dashboard] Restructure Organization Settings

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
## Description

This PR introduces a division of the Organization's "General settings" into two distinct sections: "General" and "Policies." This change aims to provide more fine-grained control over organization settings. Additionally, new sections for "Networking" and "Authentication" have been introduced with specific visibility rules for PAYG (Pay-As-You-Go)

#### Changes

1. **General Settings Split:**
   - **General:**
     - **Content:** Set the default role and workspace image, name or delete your organization.
     - **Visibility:** Available on both Enterprise and PAYG.
     - **URL Structure:** `/settings`
   - **Policies:**
     - **Content:** Restrict workspace classes, editors, and sharing across your organization.
     - **Visibility:** Available on both Enterprise and PAYG.
     - **URL Structure:** `/settings/policy`

2. **New Sections:**
   - **Networking:**
     - **Visibility:** Only on PAYG.
     - **URL Structure:** `/settings/networking`
     - **Redirection:** Enterprise users will be redirected to `/settings` if they try to access this page.
   - **Authentication:**
     - **Visibility:** Only on PAYG.
     - **URL Structure:** `/settings/auth`
     - **Redirection:** Enterprise users will be redirected to `/settings` if they try to access this page.

3. **Existing Sections:**
   - **Billing:**
     - **Visibility:** Only on PAYG (unchanged).
   - **SSO:**
     - **Visibility:** Only on Enterprise (unchanged).
   - **Git Providers:**
     - **Visibility:** Only on Enterprise (unchanged).

#### Testing

- Copy texts in each settings section
- Verify that the "General" and "Policies" sections are visible and functional on both Enterprise and PAYG accounts.
- Ensure that the "Networking" and "Authentication" sections are only visible on PAYG accounts and that Enterprise users are redirected appropriately. -- _I'm unsure, how?_
- Confirm that the existing sections ("Billing," "SSO," and "Git Providers") retain their original visibility and functionality.
- Check URL navigation for all settings sections to ensure correct routing and redirection.

## Related Issue(s)

Fixes OPM-605

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - siddhant-ofef127027b</li>
	<li><b>🔗 URL</b> - <a href="https://siddhant-ofef127027b.preview.gitpod-dev.com/workspaces" target="_blank">siddhant-ofef127027b.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - siddhant-opm-605-add-references-to-enterprise-in-the-organizational-settings-gha.25544</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-siddhant-ofef127027b%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by svenefftinge** [comment]
Doesn't look good in dark theme: (the whole enterprise pill looks strange)

Screenshot 2024-05-29 at 08 50 44

Also the new cards have a different background color than the old ones. Should that tell me something as a user?

**Comment by svenefftinge** [comment]
> the whole enterprise pill looks strange

I'm not a fan of the bg color, but the main problem seems to be the lightness/contrast of the font color on dark theme.

**Comment by Siddhant-K-code** [comment]
I think, the different background color is expected (see [figma](https://www.figma.com/design/0o3arrIcBZqVD8KfxzggcD/Promote-Enterprise-to-PAYG?node-id=168-1508&m=dev))

Regarding, Label colour, agree with your feedback. cc: @corneliusludmann

File: components/dashboard/src/components/PillLabel.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ export type PillType = "info" | "warn" | "success" | "neutral";
 
 const PillClsMap: Record<PillType, string> = {
     info: "bg-blue-50 text-blue-500 dark:bg-blue-500 dark:text-blue-100",
-    warn: "bg-orange-100 text-orange-700 dark:bg-orange-600 dark:text-orange-100",
```
Comment (svenefftinge) [inline components/dashboard/src/components/PillLabel.tsx:11]: where doe we use this?

Comment (Siddhant-K-code) [reply components/dashboard/src/components/PillLabel.tsx:11]: [`Enterprise` Pill label](https://github.com/gitpod-io/gitpod/pull/19788#issuecomment-2136667852) is coming from here only. Additionally, it is also being used in Access tokens page (unrelated to this change)

File: components/dashboard/src/teams/TeamNetworking.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+const SelfHostedCalloutCard = () => {
+    return (
+        <ConfigurationSettingsField className="bg-pk-surface-secondary">
+            <Heading3>Self-host in your cloud account</Heading3>
+            <Subheading className="mt-1">
+                Deploy the Gitpod infrastructure into your own cloud account and connect to your private network
+            </Subheading>
+
+            <div className="mt-1 flex flex-col space-y-2">
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamNetworking.tsx:56]: I know the designs have different spacing, but it wasn't intentional 🙏 

```suggestion
            <div className="mt-8 flex flex-col space-y-2">
```

File: components/dashboard/src/teams/TeamAuthentication.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            <OrgSettingsPage>
+                <div className="space-y-8">
+                    <div>
+                        <Heading2 className="flex items-center gap-4">
+                            Authentication
+                            <PillLabel type="warn" className="!px-3 !py-0.5 !text-xs border">
+                                <span className="text-sm font-semibold text-pk-content-secondary">Enterprise</span>
+                            </PillLabel>
+                        </Heading2>
+                        <Subheading className="mt-1">Manage your organization's authentication settings.</Subheading>
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamAuthentication.tsx:36]:
```suggestion
                        <Subheading className="mt-1">Manage users through single sign-on and privately authenticate with source control and image registries</Subheading>
```

File: components/dashboard/src/teams/TeamNetworking.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            <OrgSettingsPage>
+                <div className="space-y-8">
+                    <div>
+                        <Heading2 className="flex items-center gap-4">
+                            Networking
+                            <PillLabel type="warn" className="!px-3 !py-0.5 !text-xs border">
+                                <span className="text-sm font-semibold text-pk-content-secondary">Enterprise</span>
+                            </PillLabel>
+                        </Heading2>
+                        <Subheading className="mt-1">Manage your organization's networking settings.</Subheading>
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamNetworking.tsx:36]:
```suggestion
                        <Subheading className="mt-1">Self-host a single-tenant installation in your own cloud account </Subheading>
```

File: components/dashboard/src/teams/TeamNetworking.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        </ConfigurationSettingsField>
+    );
+};
+
+const VPNCard = () => {
+    return (
+        <ConfigurationSettingsField className="bg-pk-surface-secondary">
+            <Heading3>Virtual Private Network (VPN)</Heading3>
+            <Subheading className="mt-1">
+                Ensure that only your developers can access your Gitpod instance through your VPN network
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamNetworking.tsx:122]:
```suggestion
                Restrict access to your instance using your own private VPN network
```

File: components/dashboard/src/teams/TeamNetworking.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+            </Subheading>
+
+            <div className="mt-8 flex flex-col space-y-2">
+                <div className="flex flex-row gap-2 items-center text-pk-content-secondary">
+                    <CheckCircle2Icon size={20} />
+                    Meet data residency compliance requirements
+                </div>
+                <div className="flex flex-row gap-2 items-center text-pk-content-secondary">
+                    <CheckCircle2Icon size={20} />
+                    Reduce your workspace latency
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamNetworking.tsx:101]:
```suggestion
                    Reduce workspace latency and move your code closer to your data
```

File: components/dashboard/src/teams/TeamNetworking.tsx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                    <VPNCard />
+                </div>
+            </OrgSettingsPage>
+        </>
+    );
+}
+
+const SelfHostedCalloutCard = () => {
+    return (
+        <ConfigurationSettingsField className="bg-pk-surface-secondary">
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamNetworking.tsx:50]: The borders between the grey info boxes and the white seem to be different (32px vs 16px). When you swap tabs, this mean the layout feels like it's shifting. Is it possible to align those margins? 

|Grey|White|
|-|-|
|image|image|

Comment (Siddhant-K-code) [reply components/dashboard/src/teams/TeamNetworking.tsx:50]: ahh! `Collaboration and sharing` component is using outdated + inconsistent design style (un-related to this change) -- I'll fix it.

File: components/dashboard/src/teams/TeamAuthentication.tsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+            <Heading3>Private container image registry</Heading3>
+            <Subheading className="mt-1">Provide secure access to private image registries such as ECR</Subheading>
+
+            <LinkButton
+                variant="secondary"
+                className="mt-8 border border-pk-content-tertiary text-pk-content-tertiary"
+                href="https://www.gitpod.io/docs/enterprise/setup-gitpod/use-private-ecr-repos-for-workspace-images"
+                isExternalUrl={true}
+            >
+                Dcoumentation
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamAuthentication.tsx:88]: Documentation

Comment (loujaybee) [inline components/dashboard/src/teams/TeamAuthentication.tsx:88]:
```suggestion
                Documentation
```

File: components/dashboard/src/teams/TeamAuthentication.tsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+                Connect to your private source control like GitHub, BitBucket and GitLab
+            </Subheading>
+
+            <LinkButton
+                variant="secondary"
+                className="mt-8 border border-pk-content-tertiary text-pk-content-tertiary"
+                href="https://www.gitpod.io/docs/enterprise/setup-gitpod/scm-integration"
+                isExternalUrl={true}
+            >
+                Dcoumentation
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamAuthentication.tsx:108]:
```suggestion
                Documentation
```

File: components/dashboard/src/teams/TeamNetworking.tsx [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+                </div>
+            </div>
+
+            <LinkButton
+                variant="secondary"
+                className="mt-8 border border-pk-content-tertiary text-pk-content-tertiary"
+                href="https://www.gitpod.io/docs/enterprise/overview#aws-support-and-regions"
+                isExternalUrl={true}
+            >
+                Dcoumentation
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamNetworking.tsx:111]:
```suggestion
                Documentation
```

File: components/dashboard/src/teams/TeamNetworking.tsx [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+                Ensure that only your developers can access your Gitpod instance through your VPN network
+            </Subheading>
+
+            <LinkButton
+                variant="secondary"
+                className="mt-8 border border-pk-content-tertiary text-pk-content-tertiary"
+                href="https://www.gitpod.io/docs/enterprise/getting-started/networking#private-networking-configuration-highly-restrictive"
+                isExternalUrl={true}
+            >
+                Dcoumentation
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamNetworking.tsx:131]:
```suggestion
                Documentation
```

File: components/dashboard/src/teams/TeamPolicies.tsx [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+            </Subheading>
+
+            <div className="mt-6 flex flex-row space-x-2">
+                <LinkButton
+                    variant="secondary"
+                    className="border border-pk-content-tertiary text-pk-content-tertiary"
+                    href="https://www.gitpod.io/docs/configure/workspaces/workspace-classes#enterprise"
+                    isExternalUrl={true}
+                >
+                    Dcoumentation
```
Comment (loujaybee) [inline components/dashboard/src/teams/TeamPolicies.tsx:298]:
```suggestion
                    Documentation
```

File: components/dashboard/src/teams/OrgSettingsPage.tsx [hunk 13]
```diff
@@ -82,7 +83,23 @@ function getOrgSettingsMenu(params: {
...
         },
+        {
+            title: "Policies",
+            link: [`/settings/policy`],
+        },
     ];
+    if (isGitpodIo()) {
+        result.push(
+            {
+                title: "Neworking",
```
Comment (phimae) [inline components/dashboard/src/teams/OrgSettingsPage.tsx:94]: typo - should be networking @Siddhant-K-code

**Comment by Siddhant-K-code** [comment]
@svenefftinge PR is ready for review again -- I've aligned the styles with our new designs from @corygrunk.

| Light | Dark |
|--------|--------|
| image | image |

File: components/dashboard/src/user-settings/PersonalAccessTokens.tsx [hunk 14]
```diff
@@ -169,8 +169,8 @@ function ListAccessTokensView() {
         <>
             <div className="flex items-center sm:justify-between mb-4">
                 <div>
-                    <Heading2>
-                        Access Tokens{" "}
+                    <Heading2 className="flex gap-4 items-center">
+                        Access Tokens
```
Comment (Siddhant-K-code) [inline components/dashboard/src/user-settings/PersonalAccessTokens.tsx:173]: This is a minor style change for the PAT Token page to have a consistent design :)

File: components/dashboard/src/user-settings/PersonalAccessTokens.tsx [hunk 15]
```diff
@@ -169,9 +169,9 @@ function ListAccessTokensView() {
         <>
             <div className="flex items-center sm:justify-between mb-4">
                 <div>
-                    <Heading2>
-                        Access Tokens{" "}
-                        <PillLabel type="warn" className="font-semibold self-center py-0.5 px-1.5">
+                    <Heading2 className="flex gap-4 items-center">
+                        Access Tokens
+                        <PillLabel type="warn">
```
Comment (Siddhant-K-code) [inline components/dashboard/src/user-settings/PersonalAccessTokens.tsx:174]: This is a minor style change for the PAT Token page to have a consistent design (for heading + label) :)

**Review by svenefftinge — APPROVED** [review]
lgtm (minus the mentioned styling differences mentioned in slack)

**Comment by Siddhant-K-code** [comment]
> (minus the mentioned styling differences mentioned in Slack)

I addressed most of the bits now. Looking better (shared screenshots in Slack) -- will merge this. And, I'll circle back and fix, it if @corygrunk has some other suggestions.

**Pull request merged**